packages/read-model/src/Product/Listed/ListedProductViewElasticFacade.php

```diff
@@ -0,0 +1,276 @@
+            $limit
+        );
+
+        return $this->createFromProducts($accessories);
+    }
+
+    /**
+     * @param int $productId
+     * @return \Shopsys\ReadModelBundle\Product\Listed\ListedProductView[]
+     */
+    public function getAllAccessories(int $productId): array
+    {
+        $product = $this->productFacade->getById($productId);
+
+        $accessories = $this->productAccessoryFacade->getTopOfferedAccessories(
+            $product,
+            $this->domain->getId(),
+            $this->currentCustomer->getPricingGroup(),
+            null
+        );
+
+        return $this->createFromProducts($accessories);
+    }
+
+    /**
+     * @param int $categoryId
+     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData $filterData
+     * @param string $orderingModeId
+     * @param int $page
+     * @param int $limit
+     * @return \Shopsys\FrameworkBundle\Component\Paginator\PaginationResult
+     */
+    public function getFilteredPaginatedInCategory(int $categoryId, ProductFilterData $filterData, string $orderingModeId, int $page, int $limit): PaginationResult
+    {
+        $paginationResult = $this->productOnCurrentDomainFacade->getPaginatedProductsInCategory($filterData, $orderingModeId, $page, $limit, $categoryId);
+
+        return $this->createPaginationResultWithArray($paginationResult);
+    }
+
+    /**
+     * @param string $searchText
+     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData $filterData
+     * @param string $orderingModeId
+     * @param int $page
+     * @param int $limit
+     * @return \Shopsys\FrameworkBundle\Component\Paginator\PaginationResult
+     */
+    public function getFilteredPaginatedForSearch(string $searchText, ProductFilterData $filterData, string $orderingModeId, int $page, int $limit): PaginationResult
+    {
+        $paginationResult = $this->productOnCurrentDomainFacade->getPaginatedProductsForSearch($searchText, $filterData, $orderingModeId, $page, $limit);
+
+        return $this->createPaginationResultWithArray($paginationResult);
+    }
+
+    /**
+     * @param int $brandId
+     * @param string $orderingModeId
+     * @param int $page
+     * @param int $limit
+     * @return \Shopsys\FrameworkBundle\Component\Paginator\PaginationResult
+     */
+    public function getPaginatedForBrand(int $brandId, string $orderingModeId, int $page, int $limit): PaginationResult
+    {
+        $paginationResult = $this->productOnCurrentDomainFacade->getPaginatedProductsForBrand($orderingModeId, $page, $limit, $brandId);
+
+        return $this->createPaginationResultWithArray($paginationResult);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Paginator\PaginationResult $paginationResult
+     * @return \Shopsys\FrameworkBundle\Component\Paginator\PaginationResult
+     */
+    protected function createPaginationResultWithArray(PaginationResult $paginationResult): PaginationResult
```

Comment:
I don't understand just by the name what is the method doing.... could you think about a better name?

Reply:
I've tried when I implemented this but I really can't come up with a better name and I think that it might be OK like this ... do you have any ideas?

Reply:
It's already used, and my only suggestion would be `createPaginationResultWithArray`... so ok